lib/matplotlib/transforms.py

```diff
@@ -286,232 +281,216 @@ def is_unit(self):
-         (property) :attr:`y1` is the second of the pair of *y* coordinates
-         that define the bounding box.  :attr:`y1` is not guaranteed to be
-         greater than :attr:`y0`.  If you require that, use :attr:`ymax`.""")
 
-    def _get_p0(self):
+    @property
+    def p0(self):
+        """
+        (property) :attr:`p0` is the first pair of (*x*, *y*) coordinates that
+        define the bounding box.  It is not guaranteed to be the bottom-left
+        corner.  For that, use :attr:`min`.
+        """
         return self.get_points()[0]
-    p0 = property(_get_p0, None, None, """
-         (property) :attr:`p0` is the first pair of (*x*, *y*) coordinates
-         that define the bounding box.  It is not guaranteed to be the
-         bottom-left corner.  For that, use :attr:`min`.""")
 
-    def _get_p1(self):
+    @property
+    def p1(self):
+        """
+        (property) :attr:`p1` is the second pair of (*x*, *y*) coordinates that
+        define the bounding box.  It is not guaranteed to be the top-right
+        corner.  For that, use :attr:`max`.
+        """
         return self.get_points()[1]
-    p1 = property(_get_p1, None, None, """
-         (property) :attr:`p1` is the second pair of (*x*, *y*) coordinates
-         that define the bounding box.  It is not guaranteed to be the
-         top-right corner.  For that, use :attr:`max`.""")
-
-    def _get_xmin(self):
-        return min(self.get_points()[:, 0])
-    xmin = property(_get_xmin, None, None, """
-        (property) :attr:`xmin` is the left edge of the bounding box.""")
-
-    def _get_ymin(self):
-        return min(self.get_points()[:, 1])
-    ymin = property(_get_ymin, None, None, """
-        (property) :attr:`ymin` is the bottom edge of the bounding box.""")
-
-    def _get_xmax(self):
-        return max(self.get_points()[:, 0])
-    xmax = property(_get_xmax, None, None, """
-        (property) :attr:`xmax` is the right edge of the bounding box.""")
-
-    def _get_ymax(self):
-        return max(self.get_points()[:, 1])
-    ymax = property(_get_ymax, None, None, """
-        (property) :attr:`ymax` is the top edge of the bounding box.""")
-
-    def _get_min(self):
-        return [min(self.get_points()[:, 0]),
-                min(self.get_points()[:, 1])]
-    min = property(_get_min, None, None, """
-        (property) :attr:`min` is the bottom-left corner of the bounding
-        box.""")
-
-    def _get_max(self):
-        return [max(self.get_points()[:, 0]),
-                max(self.get_points()[:, 1])]
-    max = property(_get_max, None, None, """
-        (property) :attr:`max` is the top-right corner of the bounding box.""")
-
-    def _get_intervalx(self):
-        return self.get_points()[:, 0]
-    intervalx = property(_get_intervalx, None, None, """
+
+    @property
+    def xmin(self):
+        """
+        (property) :attr:`xmin` is the left edge of the bounding box.
+        """
+        return np.min(self.get_points()[:, 0])
```

Comment:
So that `nan` passes through -- which is probably the right thing to do?  (`min(np.nan, 1)` is 1, `np.min(np.nan, 1)` is np.nan.)  (and in fact needed for the various new implementations of the `contains`/`overlaps` methods)